typescript/core/src/Reader.ts

```diff
@@ -1,9 +1,10 @@
 import { getBigUint64 } from "./getBigUint64";
 
+const textDecoder = new TextDecoder();
```

Comment:
We should add a comment about why we are using a global decoder.

I'm not aware of any issue with doing this (I've long hated that the API requires a new instance each time) but it could be that the decoder holds onto some internal state for longer than we'd want?

Reply:
added a comment. The encoding seems to be the TextDecoder's only internal state